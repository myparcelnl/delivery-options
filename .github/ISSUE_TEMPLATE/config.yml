blank_issues_enabled: false
contact_links:
  - name: Try the sandbox
    url: https://myparcelnl.github.io/delivery-options/
    about: The sandbox contains all available options and allows you to build or test configurations.
  - name: Contact customer support
    url: https://www.myparcel.nl/services/contact/
    about: For issues regarding your account and shipments.
