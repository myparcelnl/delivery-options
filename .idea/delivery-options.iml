<?xml version="1.0" encoding="UTF-8"?>
<module type="WEB_MODULE" version="4">
  <component name="NewModuleRootManager">
    <content url="file://$MODULE_DIR$">
      <excludeFolder url="file://$MODULE_DIR$/dist" />
      <excludeFolder url="file://$MODULE_DIR$/.yarn" />
      <excludeFolder url="file://$MODULE_DIR$/coverage" />
      <excludeFolder url="file://$MODULE_DIR$/artifact" />
      <excludeFolder url="file://$MODULE_DIR$/.husky" />
      <excludeFolder url="file://$MODULE_DIR$/.nx" />
    </content>
    <orderEntry type="inheritedJdk" />
    <orderEntry type="sourceFolder" forTests="false" />
  </component>
</module>