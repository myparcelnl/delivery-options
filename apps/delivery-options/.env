VITE_CLASS_BASE=myparcel-delivery-options
