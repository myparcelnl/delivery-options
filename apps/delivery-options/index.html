<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta
      content="width=device-width, initial-scale=1.0"
      name="viewport" />
    <title>MyParcel Delivery Options</title>
  </head>
  <body>
    <script type="importmap">
      {
        "imports": {
          "vue": "https://unpkg.com/vue@3/dist/vue.esm-browser.js"
        }
      }
    </script>

    <div id="myparcel-delivery-options"></div>

    <script src="/src/main.ts" type="module"></script>
  </body>
</html>
