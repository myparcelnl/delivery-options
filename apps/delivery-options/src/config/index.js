export * from './appConfig';
export * from './configBus';
export * from './configurationMerger';
export * from './defaultConfiguration';
export * from './defaultStrings';
export * from './errorConfig';
export * from './eventConfig';
export * from './exports';
export * from './extraDeliveryConfig';
export * from './formConfig';
export * from './getAddress';
export * from './getConfig';
export * from './getWindowObject';
export * from './hasFilterableValue';
export * from './isPastTime';
export * from './localeConfig';
export * from './platform';
export * from './settingHasCarrierOverride';
export * from './settingHasCountryOverride';
export * from './urlConfig';
export * from './validatePlatform';
