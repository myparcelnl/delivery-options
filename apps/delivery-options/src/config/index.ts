export * from './filterConfig';
export * from './getDefaultConfigForPlatform';
export * from './getDefaultStrings';
export * from './handleDeprecatedOptions';
export * from './setConfiguration';
export * from './validateConfig';
