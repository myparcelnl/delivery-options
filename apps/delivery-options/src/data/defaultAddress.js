/* eslint-disable no-magic-numbers */
import * as ADDRESS from '../data/keys/addressKeys';
import { MYPARCEL, SENDMYPARCEL } from './keys/platformKeys';
import { countryCodes } from '../data/keys/countryCodes';

/**
 * @type {object<MyParcelDeliveryOptions.Address>}
 */
export const defaultAddress = {
  [MYPARCEL]: {
    [ADDRESS.CC]: countryCodes.NETHERLANDS,
    [ADDRESS.NUMBER]: 68,
    [ADDRESS.POSTAL_CODE]: '2514GL',
  },
  [SENDMYPARCEL]: {
    [ADDRESS.CC]: countryCodes.BELGIUM,
    [ADDRESS.CITY]: 'Antwerpen',
    [ADDRESS.NUMBER]: 16,
    [ADDRESS.POSTAL_CODE]: '2000',
  },
};
