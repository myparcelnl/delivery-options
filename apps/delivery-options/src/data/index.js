export * from './carrierFeatures';
export * from './carriers'
export * from './dates'
export * from './defaultAddress';
export * from './delivery'
export * from './keys'
export * from './pickup'
export * from './prices'
export * from './request'
