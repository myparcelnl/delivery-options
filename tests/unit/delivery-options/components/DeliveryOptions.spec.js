import * as ADDRESS from '@/data/keys/addressKeys';
import * as CONFIG from '@/data/keys/configKeys';
import {
  DISABLE_DELIVERY_OPTIONS,
  HIDE_DELIVERY_OPTIONS,
  SHOW_DELIVERY_OPTIONS,
  UPDATED_DELIVERY_OPTIONS,
  UPDATE_CONFIG_IN,
  UPDATE_DELIVERY_OPTIONS,
} from '@/config/eventConfig';
import { MYPARCEL, SENDMYPARCEL } from '@/data/keys/platformKeys';
import {
  deliveryOptionsResponseDefault,
  deliveryOptionsResponseInvalidPostalCode,
  fakeDeliveryOptionsResponse,
} from '../../../mocks/fakeDeliveryOptionsResponse';
import { createWaitableMock } from '@Tests/unit/createWaitableMock';
import { defaultAddress } from '@/data/defaultAddress';
import { defaultConfiguration } from '@/config/defaultConfiguration';
import { mockDeliveryOptions } from '@Tests/unit/delivery-options/mockDeliveryOptions';
import { waitForEvent } from '@Tests/waitForEvent';

const configWithInvalidAddress = {
  ...defaultConfiguration(SENDMYPARCEL),
  [ADDRESS.KEY]: {
    // Invalid because cc is missing
    postalCode: '1234AB',
    number: '1',
  },
};

const configWithValidAddress = {
  ...configWithInvalidAddress,
  [ADDRESS.KEY]: defaultAddress[MYPARCEL],
};

describe('DeliveryOptions.vue', () => {
  it('checks address requirements', async() => {
    expect.assertions(5);

    // Load with a valid address.
    const wrapper = mockDeliveryOptions(MYPARCEL);
    wrapper.vm.showAddressErrors = createWaitableMock(wrapper.vm.showAddressErrors);

    await waitForEvent(UPDATED_DELIVERY_OPTIONS);
    expect(wrapper.vm.hasValidAddress).toBe(true);
    expect(wrapper.vm.showAddressErrors).toBeCalledTimes(0);

    // Change address to an invalid address.
    document.dispatchEvent(
      new CustomEvent(UPDATE_DELIVERY_OPTIONS, {
        detail: configWithInvalidAddress,
      }),
    );
    await wrapper.vm.$nextTick();
    await wrapper.vm.showAddressErrors.waitToHaveBeenCalled(1);
    expect(wrapper.vm.hasValidAddress).toBe(false);

    // Change to a valid address again
    document.dispatchEvent(
      new CustomEvent(UPDATE_DELIVERY_OPTIONS, {
        detail: configWithValidAddress,
      }),
    );
    await waitForEvent(UPDATED_DELIVERY_OPTIONS);

    expect(wrapper.vm.showAddressErrors).toBeCalledTimes(1);
    expect(wrapper.vm.hasValidAddress).toBe(true);
  });

  it('shows errors on invalid address response from api', async() => {
    expect.assertions(2);
    fakeDeliveryOptionsResponse.mockImplementation(deliveryOptionsResponseInvalidPostalCode);

    const wrapper = mockDeliveryOptions({
      [ADDRESS.KEY]: {
        ...defaultAddress[MYPARCEL],
        postalCode: '12',
      },
    });
    wrapper.vm.showAddressErrors = createWaitableMock(wrapper.vm.showAddressErrors);

    await wrapper.vm.showAddressErrors.waitToHaveBeenCalled(1);
    await wrapper.vm.$nextTick();
    expect(wrapper.findByTestId('button--retry').exists()).toBe(true);

    // Change address to a valid address.
    fakeDeliveryOptionsResponse.mockImplementation(deliveryOptionsResponseDefault);
    document.dispatchEvent(new CustomEvent(UPDATE_DELIVERY_OPTIONS, {
      detail: configWithValidAddress,
    }));

    await waitForEvent(UPDATED_DELIVERY_OPTIONS);
    expect(wrapper.findByTestId('button--retry').exists()).toBe(false);
  });

  it('only shows itself when necessary', async() => {
    expect.assertions(2);
    const wrapper = mockDeliveryOptions();
    expect(wrapper.vm.hasSomethingToShow).toBe(true);

    document.dispatchEvent(new CustomEvent(UPDATE_CONFIG_IN, {
      detail: {
        [CONFIG.KEY]: {
          [CONFIG.PLATFORM]: MYPARCEL,
          [CONFIG.CARRIER_SETTINGS]: {
            postnl: {
              [CONFIG.ALLOW_DELIVERY_OPTIONS]: false,
              [CONFIG.ALLOW_PICKUP_LOCATIONS]: false,
            },
          },
        },
      },
    }));

    await wrapper.vm.$nextTick();
    expect(wrapper.vm.hasSomethingToShow).toBe(false);
  });

  it('can unselect delivery options', async() => {
    expect.assertions(1);
    const wrapper = mockDeliveryOptions();
    await waitForEvent(UPDATED_DELIVERY_OPTIONS);

    document.dispatchEvent(new Event(DISABLE_DELIVERY_OPTIONS));
    await wrapper.vm.$nextTick();

    expect(wrapper.findAll('input:checked')).toHaveLength(0);
  });

  it('hides and shows delivery options', async() => {
    expect.assertions(2);
    const wrapper = mockDeliveryOptions(SENDMYPARCEL);

    document.dispatchEvent(new Event(HIDE_DELIVERY_OPTIONS));

    // Wait for the "last" update event.
    await waitForEvent(UPDATED_DELIVERY_OPTIONS);
    expect(wrapper.vm.$el).toMatchInlineSnapshot('<!---->');

    document.dispatchEvent(new Event(SHOW_DELIVERY_OPTIONS));
    await waitForEvent(UPDATED_DELIVERY_OPTIONS);
    expect(wrapper.findChoice('delivery', 'deliver').isVisible()).toBeTruthy();
  });

  it('clears all listeners on destroy', () => {
    const wrapper = mockDeliveryOptions(SENDMYPARCEL);

    wrapper.vm.$destroy();
  });
});
