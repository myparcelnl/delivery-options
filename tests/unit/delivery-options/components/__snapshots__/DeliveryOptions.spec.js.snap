// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliveryOptions.vue hides and shows delivery options 2`] = `<!---->`;
